std/range/package.d

```diff
@@ -1475,6 +1481,15 @@ private struct ChooseResult(R1, R2)
         }
     }
 
+    void opAssign(return scope ChooseResult r) @trusted
```

Comment:
@WalterBright: Ouch.

```d
import std.range;

struct EvilRange {
    enum empty = true;
    int front;
    void popFront() @safe {}
    auto opAssign(const ref EvilRange other){
        *(cast(uint*)0xcafebabe) = 0xdeadbeef;
        return this;
    }
}

void main() @safe {

    auto c1 = choose(true, EvilRange(), EvilRange());
    auto c2 = c1;
    c1 = c2;
}
```

(Filed as https://issues.dlang.org/show_bug.cgi?id=19738.)

Reply:
Still unfixed – should we revert this commit? (I don't have time to look into this right now, sorry.)

Reply:
No, because of the other fixes in it. Just file a new PR.